### PR TITLE
Secway and janicart key suicides

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request

Adds a unique suicide to the secway key, where the user spins and then gibs.
Adds a unique suicide to the janitor cart key, where it checks your skill level and does a thematic suicide.
If you have no cleaning level, or a novice level, you immediately gib.
If you are an apprentice or journeyman you clean the tiles you walk on and then die in 5 seconds.
If you are an expert or a master you flash janicart colors and clean the tiles you walk on and then die in 10 seconds.
If you are a legendary janny though, you have a windup noise, flash janicart colors, clean tiles you walk on, gain 10u of meth in your system, and die in 15 seconds.
As an added bonus legendary jannies bwoink when their time as a janicart expires.

## Why It's Good For The Game

I was extremely disappointed to find that there was no suicide for this unique item, and decided to come up with my own idea.

Being asked why I didn't do the janicart with the secway made me think of something better, and more thematic.
I, for one, respect the janitorial arts.

## Changelog
:cl:
add: The secway key now has a unique suicide
add: The janicart key now has a uniquer suicide that respects janitorial skill
/:cl:

### CODE REVIEW

File: code/modules/vehicles/vehicle_key.dm [hunk 1]
```diff
@@ -9,6 +9,12 @@
 	desc = "A keyring with a small steel key, and a rubber stun baton accessory."
 	icon_state = "keysec"
 
+/obj/item/key/security/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] is putting \the [src] in [user.p_their()] ear and begins spinning! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	user.emote("spin")
```
Comment (ATH1909) [inline code/modules/vehicles/vehicle_key.dm]: add a shame suicide if they're somehow unable to spin

Comment (ArcaneDefence) [reply code/modules/vehicles/vehicle_key.dm]: Can't say I know how to do that

Comment (ArcaneDefence) [reply code/modules/vehicles/vehicle_key.dm]: Got that figured out.

**Comment by Dennok** [comment]
Nice but why not for janitorial key too?

**Comment by PKPenguin321** [comment]
This is pretty close to the hand drill suicide I added but with less spin

**Comment by ArcaneDefence** [comment]
I wanted to put something silly to the secway key because the secway inspires images of hopcurity, RDsec, and dufflesec. I don't really think I want to mock the jannycart key with this suicide, as using the floor buffer is genuinely good.

**Comment by Pumpkinoe** [comment]
does curator's case key have a unique suicide

**Comment by ArcaneDefence** [comment]
> 
> 
> does curator's case key have a unique suicide

No, it does not. The only thing I can think of is to have it create a display case with your head in it, but those cases set off alarms when smashed. Not really something I want to touch right now.

File: code/modules/vehicles/vehicle_key.dm [hunk 2]
```diff
@@ -9,10 +9,57 @@
...
+			user.reagents.add_reagent(/datum/reagent/drug/methamphetamine, 10) //Gotta go fast!
+			for(var/i in 1 to 150)
+				addtimer(CALLBACK(user, /atom/proc/add_atom_colour, (i % 2)? "#a245bb" : "#7a7d82", ADMIN_COLOUR_PRIORITY), i)
+			addtimer(CALLBACK(src, .proc/manual_suicide, user), 151)
+			return MANUAL_SUICIDE
+
+/obj/item/key/proc/manual_suicide(mob/living/user)
+	if(user)
+		user.remove_atom_colour(ADMIN_COLOUR_PRIORITY)
+		user.visible_message("<span class='suicide'>[user] forgot [user.p_they()] isn't actually a janicart! That's a paddlin'!</span>")
```
Comment (ATH1909) [inline code/modules/vehicles/vehicle_key.dm]: "they isn't"?

Comment (ArcaneDefence) [reply code/modules/vehicles/vehicle_key.dm]: If I'm told they is a singular pronoun I'm going to use it as a singular pronoun.
He isn't, she isn't, they isn't.
/shrug

**Comment by MrDoomBringer** [comment]
based jani suicide

**Comment by Dennok** [comment]
> > does curator's case key have a unique suicide
> 
> No, it does not. The only thing I can think of is to have it create a display case with your head in it, but those cases set off alarms when smashed. Not really something I want to touch right now.

Case can be created alarmless, see case construction, it must create alarmless case, if it PR merged.

**Pull request merged**